src/SoftKiller.jl

```diff
@@ -0,0 +1,203 @@
+using JetReconstruction
+
+"""
+    mutable struct SoftKiller
+
+Implements the SoftKiller pileup mitigation algorithm, inspired by the FastJet contrib package.
+
+The original algorithm is described in:
+https://arxiv.org/abs/1407.0408
+by Matteo Cacciari, Gavin P. Salam, Gregory Soyez
+
+This version inspired by the SoftKiller implementation in the FastJet contrib package
+Original C++ code: https://fastjet.hepforge.org/contrib/
+
+The SoftKiller algorithm divides the rapidity-phi plane into a grid of tiles and determines a dynamic
+pt threshold by finding the median of the maximum pt in each tile. Particles with pt below this threshold
+are removed from the event.
+
+# Fields
+- `_ymax::Float64`: Maximum rapidity of the grid.
+- `_ymin::Float64`: Minimum rapidity of the grid.
+- `_requested_drap::Float64`: Requested grid spacing in rapidity.
+- `_requested_dphi::Float64`: Requested grid spacing in phi.
+- `_ntotal::Int64`: Total number of tiles.
+- `_dy::Float64`: Actual grid spacing in rapidity.
+- `_dphi::Float64`: Actual grid spacing in phi.
+- `_cell_area::Float64`: Area of a single tile.
+- `_inverse_dy::Float64`: Inverse of rapidity grid spacing.
+- `_inverse_dphi::Float64`: Inverse of phi grid spacing.
+- `_ny::Int64`: Number of tiles in rapidity.
+- `_nphi::Int64`: Number of tiles in phi.
+
+# Constructors
+- `SoftKiller(rapmin::Float64, rapmax::Float64, drap::Float64, dphi::Float64)`: 
+  Construct a grid from `rapmin` to `rapmax` in rapidity, with tile sizes `drap` and `dphi`.
+- `SoftKiller(rapmax::Float64, grid_size::Float64)`: 
+  Construct a square grid from `-rapmax` to `rapmax` in rapidity, with tile size `grid_size`.
+
+"""
+mutable struct SoftKiller
+    _ymax::Float64
+    _ymin::Float64
+    _requested_drap::Float64
+    _requested_dphi::Float64
+    _ntotal::Int64
+    _dy::Float64
+    _dphi::Float64
+    _cell_area::Float64
+    _inverse_dy::Float64
+    _inverse_dphi::Float64
+    _ny::Int64
+    _nphi::Int64
+
+    
+    """
```

Comment:
```suggestion

    """
```

There is an extra line with whitespaces here which causes formatter check failure